### ISSUE TITLE
PDF Preview not working

### ISSUE THREAD

**Issue by ZeeSarp** [owncloud/core#41208]
ownCloud 10.14.0 (stable, docker image, arm64)

I added mappings to the docker-compose.yml, so my changes to the config.php are persisted:
![image](https://github.com/owncloud/core/assets/51046600/ec1cfbc7-85df-475b-98d9-bbcdaf82ca3c)
I added the required elements to the array in the config.php:
![image](https://github.com/owncloud/core/assets/51046600/fdb4ad7a-194c-44de-a285-e31e5fccf3d5)
But after starting the container, there were no pdf previews to be seen.
![image](https://github.com/owncloud/core/assets/51046600/43cf738d-fe69-450f-9db5-d8f9b0b5556a)

I also added `<policy domain="coder" rights="none" pattern="PDF" />` as a subnode of `<policymap>` to the policy.xml, but it was neither ImageMagick version 6, nor located at /etc/ImageMagick-6/policy.xml.
Found it at "/usr/local/etc/ImageMagick-7/policy.xml". 
Also, the version wasn't 3.4 but [3.7](https://gist.github.com/ZeeSarp/acf7e359105a43281d76a41c68d93383) instead.
Is the [documentation](https://doc.owncloud.com/server/10.14/admin_manual/configuration/files/previews_configuration.html#notes-for-pdf-preview-generation) outdated, or is my container weird?

Am I missing a step here?

### Steps to reproduce
1. Deploy owncloud containers
2. Add PDF provider to config.php
3. Upload pdf document

### Expected behaviour
The web UI requests and displays a preview for pdf files.

### Actual behaviour
The web UI does not request, nor does it display a preview for pdf files
![image](https://github.com/owncloud/core/assets/51046600/efecb9ab-a08d-4bd0-b54b-40c9ede39f8d)

### Server configuration
**Operating system**:
debian as docker host

**Web server:**
No idea, probably apache2

**Database:**
Also no idea.
Docker compose specifies both a container for redis and for mariadb

**PHP version:**
phpinfo()
PHP Version => 7.4.3-4ubuntu2.20

**ownCloud version:** (see ownCloud admin page)
10.14.0 

**Updated from an older ownCloud or fresh install:**
fresh install

**Where did you install ownCloud from:**
[dockerhub](https://hub.docker.com/layers/owncloud/server/10.14.0/images/sha256-e11812af68bd03912e65747181d7b067ed771e382cf14c84a9d5c4a0d95a6673?context=explore)

**Signing status (ownCloud 9.0 and above):**
"No errors have been found."

**The content of config/config.php:**
Default config.php, except for the two added elements visible in the screenshot above.

**List of activated apps:**
Can't successfully execute "php occ app:list". Produces lots of exceptions.
[gist](https://gist.github.com/ZeeSarp/808419996fcf68f23302e0e46e361cf8)

**Are you using external storage, if yes which one:** local/smb/sftp/...
no

**Are you using encryption:** yes/no
no

### Client configuration
**Browser:**
Edge

**Operating system:**
Windows 11

#### ownCloud log (data/owncloud.log)
does not exist

**Comment by phil-davis** [comment]
I have entries in "enabledPreviewProviders" like:
```
'OC\\Preview\\PDF'
```

Maybe you need to put double-backslash in the single-quoted strings to actually specify a literal backslash

https://www.php.net/manual/en/language.types.string.php#language.types.string.syntax.single
"To specify a literal single quote, escape it with a backslash (\). To specify a literal backslash, double it (\\). "

**Comment by ZeeSarp** [comment]
Good point, forgot about escaping tbh.
But according to the [php documentation](https://www.php.net/manual/en/language.types.string.php#language.types.string.syntax.single) that should not be necessary.

I tried it anyway (cahnged config.php and redeployed, logged into owncloud again, navigated to a folder with pdf and refreshed the page), but that did not seem to help.
The web page still does only send preview requests to the api for the png i uploaded, and ignores the pdf.
If I recall correctly, the web page did send request fpr pdfs as well in 10.12.
Ist there anything that i need to configure to make the web page request previews for pdfs?

**Comment by DeepDiver1975** [comment]
from my understanding you need to apply any configuration change via env variables.
editing config.php directly (even when being mounted) is not how the docker images are intended to be used.

please give it a try by setting these env vars:
- OWNCLOUD_ENABLED_PREVIEW_PROVIDERS
- OWNCLOUD_ENABLE_PREVIEWS

> I also added `<policy domain="coder" rights="none" pattern="PDF" />` as a subnode of `<policymap>` to the policy.xml, but it was neither ImageMagick version 6, nor located at /etc/ImageMagick-6/policy.xml.
> Found it at "/usr/local/etc/ImageMagick-7/policy.xml".
> Also, the version wasn't 3.4 but [3.7](https://gist.github.com/ZeeSarp/acf7e359105a43281d76a41c68d93383) instead.
> Is the [documentation](https://doc.owncloud.com/server/10.14/admin_manual/configuration/files/previews_configuration.html#notes-for-pdf-preview-generation) outdated, or is my container weird?

Docs are referring to standard Ubuntu based setups. The docker image has necessary changes already applied

**Comment by iasdeoupxe** [comment]
Most likely a duplicate of #41185 and most likely Docker image specific (PDFs ware working for me on 10.14.0)

**Comment by ZeeSarp** [comment]
Tried the environment variable approach (without manually editing the config.php and without mappings), but the underlying issue seems to be the same as in the issue iasdeoupxe mentioned.

![image](https://github.com/owncloud/core/assets/51046600/f080eb33-b6bd-44f8-8935-fc6f40977e8a)
If I understood the code correctly, the missing PDF support is the reason for the missing preview requests, as PDF is not registered in PreviewManager.php registerCoreProviders, so "oc_appconfig.core.enabledPreviewProviders" in filelist.js will never contain the PDF mime type and thus not try to load a preview.

Would it be possible to output a warning or an error when a provider that is specified in config.php is not available?
And is there a way to enable pdf support or to obtain a version of ImageMagick with pdf support?

**Comment by ZeeSarp** [comment]
Nevermind!
Just found out that the imagemagick lib requires ghostscript in order to convert PDF files.
In the [owncloud php base image](https://github.com/owncloud-docker/php/blob/master/v20.04/Dockerfile.multiarch) this is explicitly uninstalled. 
![image](https://github.com/owncloud/core/assets/51046600/b3fe788d-437e-486d-9482-dfd74be3c596)

Wasn't too sure if ghostscript depends on gsfonts and ffmpeg, so I installed all of them.
Now I have 231 enabled providers instead of 230 (PDF is one of them) and the previews work:
![image](https://github.com/owncloud/core/assets/51046600/d861ec95-3e3b-4197-a4bb-d4783b800a76)
Could this be added to the documentation?

As far as I am concerned this issue can be closed.
Thank you for your help!

**Comment by iasdeoupxe** [comment]
> Could this be added to the documentation?

The previous posting sounds to me more like you should raise an issue over at https://github.com/owncloud-docker/php/issues as this looks really Docker specific.

Otherwise creating an issue at https://github.com/owncloud/docs-server/issues would be required.

**Comment by ZeeSarp** [comment]
Thanks, will do

**Comment by pako81** [comment]
Should be fixed by https://github.com/owncloud-docker/php/pull/164. Thanks for reporting.

**Comment by jnweiger** [comment]
@ZeeSarp We deployed fixed docker images for 10.13.4 and 10.14.0. Please try e.g.  

```
docker pull owncloud/server:10.14.0
docker run  -ti -p 8080:8080 -e 'OWNCLOUD_ENABLED_PREVIEW_PROVIDERS=OC\Preview\PDF,OC\Preview\SGI,OC\Preview\Heic,OC\Preview\PNG,OC\Preview\JPEG,OC\Preview\WEBP,OC\Preview\GIF,OC\Preview\BMP,OC\Preview\XBitmap,OC\Preview\MP3,OC\Preview\TXT,OC\Preview\MarkDown,OC\Preview\SVG'  owncloud/server:10.14.0
```
All kinds of previews work nicely for me.
